kitsune/wiki/utils.py

```diff
@@ -133,28 +133,41 @@ def _active_contributors_id(from_date, to_date, locale, product):
     return set(list(editors) + list(reviewers))
 
 
-def get_featured_articles(product=None):
+def get_featured_articles(product=None, locale=settings.WIKI_DEFAULT_LANGUAGE):
     """Returns 4 random articles from the most visited.
 
     If a product is passed, it returns 4 random highly visited articles.
     """
-    featured = (
+    visits = (
         WikiDocumentVisits.objects.filter(period=LAST_7_DAYS)
         .exclude(
             document__products__slug__in=settings.EXCLUDE_PRODUCT_SLUGS_FEATURED_ARTICLES
         )
         .order_by("-visits")
-        .select_related("document")
+        .select_related('document')
     )
 
     if product:
-        featured = featured.filter(document__products__in=[product.id])
+        visits = visits.filter(document__products__in=[product.id])
+
+    visits = visits[:10]
+    documents = []
+
+    if locale == settings.WIKI_DEFAULT_LANGUAGE:
+        for visit in visits:
+            documents.append(visit.document)
+    else:
+        # prefretch localised documents to avoid n+1 problem
+        visits = visits.prefetch_related(
+            Prefetch('document__translations', queryset=Document.objects.filter(locale=locale))
+        )
 
-    # Limit this to 10 entries
-    if featured.count() < 10:
-        return featured
+        for visit in visits:
+            translation = visit.document.translations.first()
+            if not translation:
+                continue
+            documents.append(translation)
 
-    # This returns a WikiDocumentVisits object and not the actual Wiki doc.
-    # The wiki doc it's under .document which is prefetched b/c of select_related
-    featured = featured[:10]
-    return random.sample(featured, 4)
+    if len(documents) <= 4:
+        return documents
+    return random.sample(documents, 4)
```

Comment:
Let's add a check in the templates to hide completely featured articles (if it doesn't exist already) in case there aren't any.